src/main/java/com/google/api/generator/gapic/composer/rest/HttpJsonServiceStubClassComposer.java

```diff
@@ -0,0 +1,508 @@
+                  .setClazz(ProtoRestSerializer.class)
+                  .setGenerics(method.inputType().reference())
+                  .build());
+
+      VariableExpr serializerVarExpr =
+          VariableExpr.withVariable(
+              Variable.builder().setName("serializer").setType(serializerVarType).build());
+
+      Expr serializerAssignExpr =
+          AssignmentExpr.builder()
+              .setVariableExpr(serializerVarExpr.toBuilder().setIsDecl(true).build())
+              .setValueExpr(
+                  MethodInvocationExpr.builder()
+                      .setStaticReferenceType(
+                          FIXED_REST_TYPESTORE.get(ProtoRestSerializer.class.getSimpleName()))
+                      .setMethodName("create")
+                      .setReturnType(serializerVarType)
+                      .build())
+              .build();
+
+      serializerExpr = serializerVarExpr;
+
+      bodyExprs.add(serializerAssignExpr);
+    } else {
+      serializerExpr =
+          MethodInvocationExpr.builder()
+              .setMethodName("create")
+              .setStaticReferenceType(
+                  FIXED_REST_TYPESTORE.get(ProtoRestSerializer.class.getSimpleName()))
+              .build();
+      if (httpBindingFieldNames.isEmpty()) {
+        returnExpr = ValueExpr.createNullExpr();
+      }
+    }
+
+    VariableExpr requestVarExpr =
+        VariableExpr.withVariable(
+            Variable.builder().setType(method.inputType()).setName("request").build());
+
+    for (String httpBindingFieldName : httpBindingFieldNames) {
+      // Handle foo.bar cases by descending into the subfields.
+      MethodInvocationExpr.Builder requestFieldGetterExprBuilder =
+          MethodInvocationExpr.builder().setExprReferenceExpr(requestVarExpr);
+      String[] descendantFields = httpBindingFieldName.split("\\.");
+      for (int i = 0; i < descendantFields.length; i++) {
+        String currFieldName = descendantFields[i];
+        String bindingFieldMethodName =
+            String.format("get%s", JavaStyle.toUpperCamelCase(currFieldName));
+        requestFieldGetterExprBuilder =
+            requestFieldGetterExprBuilder.setMethodName(bindingFieldMethodName);
+        if (i < descendantFields.length - 1) {
+          requestFieldGetterExprBuilder =
+              MethodInvocationExpr.builder()
+                  .setExprReferenceExpr(requestFieldGetterExprBuilder.build());
+        }
+      }
+
+      MethodInvocationExpr requestBuilderExpr = requestFieldGetterExprBuilder.build();
+
+      ImmutableList.Builder<Expr> paramsPutArgs = ImmutableList.builder();
+      if (fieldsVarExpr != null) {
+        paramsPutArgs.add(fieldsVarExpr);
+      }
+      paramsPutArgs.add(
+          ValueExpr.withValue(
+              StringObjectValue.withValue(JavaStyle.toLowerCamelCase(httpBindingFieldName))));
+      paramsPutArgs.add(requestBuilderExpr);
+
+      if (fieldsVarExpr == null) {
+        Expr paramsPutExpr =
```

Comment:
Perhaps I'm missing something, but this looks the same as the if-block. Can we move this assignment out, and leave just the `returnExpr` / `bodyExprs.add()` logic inside?

Reply:
I guess centennially they were different, as since one of them was missing setReturnType(), but the one with setReturnType matches both cases, so moved it out.